xml/hwbp_registers.xml

```diff
@@ -3,9 +3,15 @@
     and Debug Mode to prevent untrusted user code from causing entry into Debug
     Mode without the OS's permission.
 
-    In this section XLEN means MXLEN when in M-mode, and DXLEN when in Debug Mode.
-    Note that this makes several of the fields in \RcsrTdataOne move around based
-    on the current execution mode and value of MXLEN.
+    In this section XLEN means MXLEN when in M-mode, and DXLEN when in Debug
+    Mode. On systems where those values of XLEN can differ, this is handled
+    as follows.
+    Fields retain their values regardless of XLEN, which only affects where
+    in the register these fields appear (e.g. \FcsrTdataOneType). Some fields
+    are wider when XLEN is 64 than when it is 32 (e.g.
+    \FcsrTextraThirtytwoSvalue). The high bits in such fields retain their
+    value but are not readable when XLEN is 32. A write to a field when XLEN
+    is 32 clears any inaccessible bits in that field.
```

Comment:
This is per-field?  Clearly, if you `csrrw` when XLEN=32 then it clears the upper 32b (unless, perhaps, there are some fields that only exist if XLEN=64 in which case there was no write to that field).  What about these cases:
- In XLEN=32, you `csrrs mcontrol` and only change sizelo?  Does sizehi (which is inaccessible) get cleared?
- In XLEN=32, you `csrrs textra32` and only change svalue.  Do the upper 7 bits of mhvalue not get cleared?

(Just trying to poke holes in things to make sure they're robust)

Reply:
How about:
A modification of a register when XLEN is 32 clears any inaccessible bits in that register.

Reply:
Perfect.  That seems straightforward to specify, to implement in hardware, and for software/debuggers to understand.  I'm not sure anything would be gained by keeping the other state around in some unmodified fields.

Reply:
I think this looks good. Apologies again for throwing a spanner in the works so late in the day.